packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift/table.ts

```diff
@@ -0,0 +1,84 @@
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as redshift from '@aws-cdk/aws-redshift';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import * as customresources from '@aws-cdk/custom-resources';
+import { Construct } from 'constructs';
+import { RedshiftColumn } from '../redshift-table';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+ * Properties for configuring a Redshift table that Firehose will load data into.
+ */
+export interface FirehoseRedshiftTableProps {
+  /**
+   * The Redshift cluster to deliver data to.
+   */
+  readonly cluster: redshift.ICluster;
+
+  /**
+   * The secret containing credentials to a Redshift user with administrator privileges.
+   */
+  readonly adminUser: secretsmanager.ISecret;
+
+  /**
+   * The database containing the desired output table.
+   */
+  readonly database: string;
+
+  /**
+   * The table columns that the source fields will be loaded into.
+   */
+  readonly tableColumns: RedshiftColumn[];
+}
+
+/**
+ * Custom resource to create the Redshift table where Firehose will load data.
+ */
+export class FirehoseRedshiftTable extends CoreConstruct {
```

Comment:
I am struggling to come up with a better name for this and `FirehoseRedshiftUser`. I think there will be some confusion with the following names:

1. `FirehoseRedshiftTable` - custom resource to create the table if needed
2. `FirehoseRedshiftUser` - custom resource to create the user if needed
3. `RedshiftTable` - the actual delivery stream destination
4. `RedshiftUser` - the class representing a user that a customer will pass into the constructor props

The relationship between 1 and 3 is not the same as the relationship between 2 and 4, even though the names have the same relationship. 

I think I don't like the prefix 'Firehose' for 1&2 since it seems to imply that Firehose has some ownership of the Table and User. 1&2 are really redshift database resources created by the CDK for Firehose to use. 

This is a very roundabout way of saying: I think 1&2 should be called something like `GeneratedRedshiftX` `CreatedRedshiftX`, `RedshiftXGenerator`, `RedshiftXBuilder`. I am also open to a different model here.